docs/cs-for-examples.md

```diff
@@ -0,0 +1,275 @@
+# Coding standard for examples
+
+Examples need to be clear and simple, but should show the possibilities and
+usage of the functions used.  Only use OOP features where you would like to
+present them, use simple functions in other areas.
+
+Examples should generally conform to the [PEAR coding standards][pear-cs].
+
+## Requirements
+
+- When appropriate, use superglobals
+- Never generate PHP errors (`E_ALL|E_STRICT` friendly)
+- Be short and generic
+- Follow the PEAR coding standards
+
+## Program listing roles  (`<programlisting role="xxx">`)
+   
+PHP examples should always be in `<programlisting role="php">`. Only
+PHP examples should have this role.  Other possible roles are:
+   
+ - c           (C code) 
+ - html        (100% XHTML)
+ - php         (Some PHP)
+ - shell       (commandline, bash, etc)
+ - sql         (SQL statements)
+ - apache-conf (Apache)
+
+## Titles
+
+When appropriate, it's encouraged to include the function name in
+the title, for example:
+
+```xml
+<title>A <function>strlen</function> example</title>
+```
+
+## Code placement
+
+The contents start at column/row 0 in the example.  For example, this
+means your example's content will be flush against the `<![CDATA[` tag.
+
+## PHP tags
+
+Always use long PHP tags (`<?php`) and never short tags (`<?` or `<?=`).
+
+## CDATA
+
+Always use `<![CDATA[ ... ]]>` as this increases the readability of
+the examples.  For example, you literally write `<` instead of `&lt;`
+inside of CDATA.  Nothing in CDATA is parsed, it's taken literally.
+So, you cannot use links, dev-comments, `<function>`, etc.
+
+## Deprecated code
+
+Do not use aliases or deprecated syntax.
+   
+## Use of newer PHP features
+
+If an example uses features, such as arguments specific to a newer
+version of PHP, add a comment that mentions this.  For example:
+
+```php   
+   // Second argument was added in PHP 4.1.0
+   foo('bar', 'baz');
+```
+   
+If appropriate, show examples that work in older versions of PHP but
+do not use reserved function names.  For example, a PHP 4.2.3 version
+of `file_get_contents()` should not be named `file_get_contents()`.
```

Comment:
Comments in examples are really annoying for translators actually. As we cannot "just" copy/paste the example.

Moreover, I would argue it should be split into 2 examples with two distinct titles.

Reply:
Rewrote this to suggest the version info should be in the title of the example.